vignettes/DataPipeline.Rmd

```diff
@@ -22,132 +22,528 @@ knitr::opts_chunk$set(
+As an overview, clinical and operational data are analyzed using the `{gsm}` R package, which generate standardized outputs and reports suitable for use in various RBQM activities. The Appendices of this document provide detailed data input and output specifications for `{gsm}`. Below is a figure that depicts the full data pipeline, including both the Analysis and Reporting sections of the process.
 
-# KRI Data Pipeline
-
-![](data-pipeline.PNG){width="125%"}
+![](data-model.PNG){width="100%"}
 
-TODO: Update/Delete Graphic
+# KRI Data Pipeline
 
 In the context of clinical research, a KRI is a measure of risk associated with the conduct of a clinical trial. Examples of KRIs include the rate of adverse events or amount of missing data at a site or across sites. Defining and deploying KRIs during study start-up allows sponsors to continually monitor risks to the integrity of the trial and take corrective actions accordingly.
 
 The default KRI data pipeline begins with one or more clinical datasets related to the KRI. An optional participant-level subset is then applied to each dataset before aggregating by participant to quantify the KRI at the participant level. This aggregated dataset is then further summarized by site to obtain site-level metrics.
 
 ## Generating Input Data
 
-TODO: Update to discuss new framework
-
-Generating input data is outside the scope of the assessment pipeline. The specifications for input data (i.e., `dfInput`) are designed to be flexible enough so that this data frame can be generated from multiple clinical data standards (e.g., Raw, rawplus, or CDISC-compliant ADaM). Most assessment workflows have `Map()` functions that can be used to generate input data. 
+Generating input data is outside the scope of the assessment pipeline. The specifications for input data (i.e., `dfInput`) are designed to be flexible enough so that this data frame can be generated from multiple clinical data standards (e.g., Raw, rawplus, or CDISC-compliant ADaM). In order to generate input data, the user uses the data frames outputted by the `data_mappings.yaml` file along with the `Input_Rate()` function to generate the `dfInput` needed for a particular KRI.
 
-For example, the Adverse Event assessment has two `Map()` functions:
 
--   `AE_Map_Raw()`: creates input data from rawplus data
+This `data_mapping.yaml` file is provided for convenience, but may not work for all clinical studies. When no relevant data.frame is produced with the `data_mapping.yaml`, the user can use the `RunQuery()` function to write a SQL query that produces the properly mapped and filtered dataframe. Details on this can be found in the [Cookbook Vignette](https://github.com/Gilead-BioStats/gsm/wiki/Cookbook-Vignette).
```

Comment:
for me, this just goes to the Cookbook-Vignette page on the gsm wiki, which will need to be updated with the new cookbook (and all other vignettes) at some point, so thanks for catching that.

Reply:
`devtools::install(build_vignettes = TRUE)` results in the proper link to a local copy of the vignette, and that blows my mind 🙃

Reply:
I *think* this should still be updated to point to Cookbook.html directly.

```suggestion
This `data_mapping.yaml` file is provided for convenience, but may not work for all clinical studies. When no relevant data.frame is produced with the `data_mapping.yaml`, the user can use the `RunQuery()` function to write a SQL query that produces the properly mapped and filtered dataframe. Details on this can be found in the [Cookbook Vignette](Cookbook.html).
```